src/Api/QueryParameterValidator/QueryParameterValidator.php

```diff
@@ -72,4 +80,76 @@ public function validateFilters(string $resourceClass, array $resourceFilters, a
             throw new FilterValidationException(array_merge(...$errorList));
         }
     }
+
+    /**
+     * @param array<string, array<string, mixed>> $filterDescription
+     */
+    private static function getCollectionFormat(array $filterDescription): string
+    {
+        return $filterDescription['openapi']['collectionFormat'] ?? $filterDescription['swagger']['collectionFormat'] ?? 'csv';
+    }
+
+    /**
+     * @param array<string, mixed> $queryParameters
+     *
+     * @throws \InvalidArgumentException
+     */
+    private static function iterateValue(string $name, array $queryParameters, string $collectionFormat = 'csv'): \Generator
+    {
+        $candidates = array_filter(
+            $queryParameters,
+            static fn (string $key) => $key === $name || "{$key}[]" === $name,
+            \ARRAY_FILTER_USE_KEY
+        );
+
+        foreach ($candidates as $key => $value) {
+            $values = self::getValue($value, $collectionFormat);
+            foreach ($values as $v) {
+                yield [$key => $v];
+            }
+        }
```

Comment:
can we do the array_filter within the loop to save some iterations

Reply:
You're right an `if ...` statement should do the trick